rpc/jsonrpc/receipts/receipts_generator.go

```diff
@@ -349,6 +350,14 @@ func (g *Generator) GetReceipts(ctx context.Context, cfg *chain.Config, tx kv.Te
 		}
 	}
 
+	// PIP-74: state sync addition
+	txs := block.Transactions()
+	if len(txs) > 0 {
```

Comment:
It's possible that there are only state-sync transactions in a block and in those cases this check won't work. I think it's better to check for every block for presence of state-sync or if bor config is available, every sprint end block.